### PR TITLE
#feat: 1418 support headers in grpc tests

### LINKED ISSUE #1418: Support headers in gRPC APIs
We recently started using Microcks for our gRPC APIs and really love the functionality and extensibility. 
One feature we are currently missing for gRPC APIs are headers. Since headers (or in gRPC lingo: gRPC Metadata) are used for authentication, custom headers, tracing or other internal usages, it would be great to support headers in mocks and especially in tests. 

I currently see the following points that would need to be addressed to fully support headers for gRPC APIs: 
- API Testing: Pass headers specified in gRPC Test Runs through to the gRPC call
- API Mocking:  Make headers available in `SCRIPT_DISPATCHER` (currently, it is always an empty list)

It would be great to see this feature in microcks! I am also happy to support here and work on a PR 🙂

### PR DESCRIPTION
**Pull request by anika-apel**
### Description

- Add headers specified in Requests and Test Run to test invocations

### Related issue(s)

Resolves #1418 

### Notes
During the implementation, I noticed that all test runners share the same code to collect the headers and refactored it into one method in `TestRunnerCommons`. If that is not wanted, let me know and I can revert this change.

### CODE REVIEW

File: webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java [hunk 1]
```diff
@@ -60,6 +70,36 @@
  */
 public class GrpcTestRunner extends AbstractTestRunner<HttpMethod> {
 
+   /* Call Option used to pass gRPC Metadata from client invocation to header client interceptor */
+   public static final CallOptions.Key<Metadata> METADATA_CUSTOM_CALL_OPTION = CallOptions.Key
+         .createWithDefault("request-metadata", null);
```
Comment (lbroudoux) [inline webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java:75]: Is `request-metadata` a standard stuff? Don't we have a Java constant for that in the `grpc-java` library?

File: webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java [hunk 2]
```diff
@@ -220,6 +269,17 @@ public void checkServerTrusted(X509Certificate[] certs, String authType) {
       return results;
    }
 
+   private static Metadata convertHeadersToMetadata(Set<Header> headers) {
+      Metadata metadata = new Metadata();
+      for (Header header : headers) {
+         log.info("Adding header {} to request", header.getName());
```
Comment (lbroudoux) [inline webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java:275]: Maybe `INFO` level is a bit high here as these are details. Could you lower this to `DEBUG`?

**Review by lbroudoux — CHANGES_REQUESTED** [review]
It looks good! (great initiative moving headers collection to `TestRunnerCommons` BTW).
I see in the CI that build is failing due to integration tests on GRPC mocks. Are your investigating if it's a side effect?

File: webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java (hunk 2 above)
Comment (anika-apel) [reply webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java]: Absolutely. Additionally, it was an unnecessary logging message only meant for debugging purposes for me during development. I removed it completely

File: webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java (hunk 1 above)
Comment (anika-apel) [reply webapp/src/main/java/io/github/microcks/util/grpc/GrpcTestRunner.java]: No there is no standard. Actually any string can be used as a name here. It should just describe the key. I moved it into a constant to highlight the intention of the name.

**Review by lbroudoux — APPROVED** [review]
Looks super good to me! Thanks!

**Pull request merged**